tests/test_compat.py

```diff
@@ -0,0 +1,125 @@
+import databricks_cli.instance_pools.api
+import databricks_cli.jobs.api
+import databricks_cli.libraries.api
+import databricks_cli.pipelines.api
+import databricks_cli.repos.api
+import databricks_cli.runs.api
+import databricks_cli.secrets.api
+import databricks_cli.tokens.api
+import databricks_cli.unity_catalog.api
+import databricks_cli.workspace.api
+
+
+def collect_argspecs(modules):
+    return {
+        func.__module__ + "." + func.__qualname__: inspect.getfullargspec(func).args
+        for module in modules
+        for (_, clazz) in inspect.getmembers(module, predicate=inspect.isclass)
+        for (_, func) in inspect.getmembers(clazz, predicate=inspect.isfunction)
+
+        # Ignore functions that are defined outside the specified module.
+        if module.__name__ == func.__module__
+    }
+
+
+# pylint: disable=superfluous-parens
+def _test_compatibility(current_argspecs, existing_argspecs):
+    removed_functions = []
+    incompatible_functions = []
+    for existing_name, existing_args in existing_argspecs.items():
+        if existing_name not in current_argspecs:
+            removed_functions.append(existing_name)
+            continue
+
+        # New arguments may only be appended to ensure compatibility
+        # when functions are called with positional parameters.
+        current_args = current_argspecs[existing_name]
+        if (
+            len(current_args) < len(existing_args)
+            or current_args[0 : len(existing_args)] != existing_args
+        ):
+            incompatible_functions.append(existing_name)
+            continue
+
+    if removed_functions:
+        print("Removed functions:")
+        for fn in removed_functions:
+            print(" * " + fn)
+        assert len(removed_functions) == 0
+
+    if incompatible_functions:
+        print("Incompatible functions:")
+        for fn in incompatible_functions:
+            current_args = current_argspecs[fn]
+            existing_args = existing_argspecs[fn]
+            print(" * " + fn + ": " + str(existing_args) + " -> " + str(current_args))
+        assert len(incompatible_functions) == 0
+
+
+@unittest.skipIf(six.PY2, reason='The `inspect` package has significantly changed in Python 3.')
+def test_compatibility():
+    api_packages = [
+        databricks_cli.cluster_policies.api,
+        databricks_cli.clusters.api,
+        databricks_cli.dbfs.api,
+        databricks_cli.groups.api,
+        databricks_cli.instance_pools.api,
+        databricks_cli.jobs.api,
+        databricks_cli.libraries.api,
+        databricks_cli.pipelines.api,
+        databricks_cli.repos.api,
+        databricks_cli.runs.api,
+        databricks_cli.secrets.api,
+        databricks_cli.tokens.api,
+        databricks_cli.unity_catalog.api,
+        databricks_cli.workspace.api,
+    ]
+
+    current_argspecs = collect_argspecs([databricks_cli.sdk.service] + api_packages)
+
+    test_compatibility_file = os.getenv('TEST_COMPATIBILITY_FILE', default=None)
```

Comment:
Why not determining this dynamically? it looks like a hack and pain to maintain.

Why not checking out `main` branch in a test and discovering argspecs from there? 

* you can either use `ast` package, though `collect_argspecs` would have to be rewritten.
* or make that checkout quack like a python sub-package, though `sys.path.modules` or something, so that `collect_argspecs` is only slightly modified.

Reply:
I like option 2. I'll give it a shot.